tutorials/using-cloudbuild-with-webhooks/index.md

```diff
@@ -0,0 +1,256 @@
+
+Up until now, you did not execute any code contained in the Git repository. In this section, you execute the repository’s Terraform code.
+1. Create a Storage Bucket to store terraform state.
+    1. Go to the [Cloud Storage page](https://console.cloud.google.com/storage/browser).
+    2. Click on **Create bucket**.
+    3. In the Name your bucket field enter “tf-state-<project\_id>” replacing the <project\_id> with your actual project\_id.
+    4. Click on the **create** button.
+2. Modify the inline Build config as shown below. \
+    ```
+    steps:
+      - name: gcr.io/cloud-builders/git
+        args:
+          - '-c'
+          - 'git clone https://gitlab-token:$$GITLAB_TOKEN@${_REPO_URL} repo'
+        entrypoint: bash
+        secretEnv:
+          - GITLAB_TOKEN
+      - name: hashicorp/terraform
+        args:
+          - init
+          - '-backend-config=bucket=${_TF_BACKEND_BAUCKET}'
+          - '-backend-config=prefix=${_TF_BACKEND_PREFIX}'
+        dir: repo
+      - name: hashicorp/terraform
+        args:
+          - plan
+          - '-var=project=${PROJECT_ID}'
+          - '-out=/workspace/tfplan-$BUILD_ID'
+        dir: repo
+      - name: hashicorp/terraform
+        args:
+          - apply
+          - '-auto-approve'
+          - /workspace/tfplan-$BUILD_ID
+        dir: repo
+    substitutions:
+      _GIT_REPO: $(body.project.git_http_url)
+      _REPO_URL: '${_GIT_REPO##https://}'
+      _TF_BACKEND_BAUCKET: 'tf-state-${PROJECT_ID}'
+      _TF_BACKEND_PREFIX: tf-state-prefix
+    availableSecrets:
+      secretManager:
+        - versionName: 'projects/${PROJECT_ID}/secrets/gitlab-token/versions/latest'
+          env: GITLAB_TOKEN
+    ```
+3. Trigger a test push event in your GitLab repository.
+4. Go to [Cloud Storage](https://console.cloud.google.com/storage/browser) and observe that there is a bucket called `test-bucket-[project_id]`.
+
+Now, whenever you push changes to the Terraform repository, you also trigger the Cloud Build pipeline which deploys changes to your project. If you decide to add another Terraform project to the same pipeline, you only need  to add a webhook to the repository.
+
+### Cleaning up
+You can clean up all the resources created in this tutorial by [shutting down the project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#shutting_down_projects). However, if you used an existing project that contains resources other than the one created in this tutorial, you should remove the resources one by one instead:
+
+1. In Cloud Shell, delete the cloud build trigger: \
+`gcloud beta builds triggers delete webhook-trigger`
+2. Delete the Cloud Secrets manager: \
+`gcloud secrets delete webhook-trigger-secret`
+`gcloud secrets delete gitlab-token`
+3. Delete the IAM policy: \
+`gcloud projects remove-iam-policy-binding <project_id_or_number> --member=serviceAccount:<project_number>@cloudbuild.gserviceaccount.com --role=roles/secretmanager.secretAccessor`
```

Comment:
can you replace the <project_id_or_number> by an environment variable here?

Reply:
+